libutils/logging.c

```diff
@@ -339,6 +351,26 @@ void VLog(LogLevel level, const char *fmt, va_list ap)
     char *msg = StringVFormat(fmt, ap);
     char *hooked_msg = NULL;
 
+    if (logging_into_buffer &&
+        (level >= min_log_into_buffer_level) && (level <= min_log_into_buffer_level))
+    {
+        assert(log_buffer != NULL);
+
+        if (log_buffer == NULL)
+        {
+            /* Should never happen. */
+            Log(LOG_LEVEL_ERR,
+                "Attempt to log a message to an unitialized log buffer, discarding the message");
```

Comment:
I think we should consider adding a `LOG_LEVEL_PANIC` or something similar in the logging API, for messages like this. This is so that the user would understand that a bug caused this message, and not the user him/her self. But that is out of scope for this PR.

Reply:
Yeah, we have `ProgrammingError()` for those cases, but it aborts the program IIRC. Anyway, that's a bit out of scope of this ticket/PR.